Error: The terraform-provider-azurerm_v3.7.0_x5 plugin crashed! when update azurerm_kubernetes_cluster

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.1.9

### AzureRM Provider Version

3.7.0

### Affected Resource(s)/Data Source(s)

azurerm_kubernetes_cluster

### Terraform Configuration Files

```hcl
resource "azurerm_kubernetes_cluster" "main" {
  ...

  lifecycle {
    ignore_changes = [
      private_cluster_public_fqdn_enabled,
      run_command_enabled,
    ]
  }
}
```


### Debug Output/Panic Output

```shell
https://gist.github.com/nigelis/ef737a98ec2b3a7cf956435b2d5ee0a2
```


### Expected Behaviour

The Kubernetes cluster is updated successfully.

### Actual Behaviour

Crash.

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

I have the same error:

```
│ The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵

Stack trace from the terraform-provider-azurerm_v3.7.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x46992d5]

goroutine 98 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/containers.resourceKubernetesClusterUpdate(0xc0001c0d00, {0x54f1ba0?, 0xc000409c00})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/containers/kubernetes_cluster_resource.go:1372 +0x1755
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc000d00450?, {0x6568b08?, 0xc00267b1c0?}, 0x24?, {0x54f1ba0?, 0xc000409c00?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:363 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000868a80, {0x6568b08, 0xc00267b1c0}, 0xc00127b930, 0xc0000a5f80, {0x54f1ba0, 0xc000409c00})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:475 +0x53a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0003755f0, {0x6568b08, 0xc00267b1c0}, 0xc001376230)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/grpc_provider.go:977 +0xd8a
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0000a5180, {0x6568bb0?, 0xc0018be1e0?}, 0xc002088000)
	github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:603 +0x2ad
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x5a59660?, 0xc0000a5180}, {0x6568bb0, 0xc0018be1e0}, 0xc00150a060, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000221180, {0x6579c50, 0xc00017a900}, 0xc0019f8000, 0xc000be1b90, 0xa488860, 0x0)
	google.golang.org/grpc@v1.39.0/server.go:1292 +0xc6f
google.golang.org/grpc.(*Server).handleStream(0xc000221180, {0x6579c50, 0xc00017a900}, 0xc0019f8000, 0x0)
	google.golang.org/grpc@v1.39.0/server.go:1617 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.39.0/server.go:940 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.39.0/server.go:938 +0x28a

Error: The terraform-provider-azurerm_v3.7.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

--- a/internal/services/containers/kubernetes_cluster_resource.go
+++ b/internal/services/containers/kubernetes_cluster_resource.go
@@ -1370,8 +1370,7 @@ func resourceKubernetesClusterUpdate(d *pluginsdk.ResourceData, meta interface{}
 	if d.HasChange("run_command_enabled") {
 		updateCluster = true
 		if existing.ManagedClusterProperties.APIServerAccessProfile == nil {
-			existing.ManagedClusterProperties.APIServerAccessProfile = &containerservice.ManagedClusterAPIServerAccessProfile{
-			}
+			existing.ManagedClusterProperties.APIServerAccessProfile = &containerservice.ManagedClusterAPIServerAccessProfile{}
 		}
 		existing.ManagedClusterProperties.APIServerAccessProfile.DisableRunCommand = utils.Bool(!d.Get("run_command_enabled").(bool))
 	}

--- a/internal/services/containers/kubernetes_cluster_resource.go
+++ b/internal/services/containers/kubernetes_cluster_resource.go
@@ -1369,13 +1369,11 @@ func resourceKubernetesClusterUpdate(d *pluginsdk.ResourceData, meta interface{}
 
 	if d.HasChange("run_command_enabled") {
 		updateCluster = true
-		if existing.ManagedClusterProperties.APIServerAccessProfile != nil {
-			existing.ManagedClusterProperties.APIServerAccessProfile.DisableRunCommand = utils.Bool(!d.Get("run_command_enabled").(bool))
-		} else {
+		if existing.ManagedClusterProperties.APIServerAccessProfile == nil {
 			existing.ManagedClusterProperties.APIServerAccessProfile = &containerservice.ManagedClusterAPIServerAccessProfile{
-				DisableRunCommand: utils.Bool(!d.Get("run_command_enabled").(bool)),
 			}
 		}
+		existing.ManagedClusterProperties.APIServerAccessProfile.DisableRunCommand = utils.Bool(!d.Get("run_command_enabled").(bool))
 	}
 
 	if d.HasChange("auto_scaler_profile") {

--- a/internal/services/containers/kubernetes_cluster_resource.go
+++ b/internal/services/containers/kubernetes_cluster_resource.go
@@ -1369,7 +1369,13 @@ func resourceKubernetesClusterUpdate(d *pluginsdk.ResourceData, meta interface{}
 
 	if d.HasChange("run_command_enabled") {
 		updateCluster = true
-		existing.ManagedClusterProperties.APIServerAccessProfile.DisableRunCommand = utils.Bool(!d.Get("run_command_enabled").(bool))
+		if existing.ManagedClusterProperties.APIServerAccessProfile != nil {
+			existing.ManagedClusterProperties.APIServerAccessProfile.DisableRunCommand = utils.Bool(!d.Get("run_command_enabled").(bool))
+		} else {
+			existing.ManagedClusterProperties.APIServerAccessProfile = &containerservice.ManagedClusterAPIServerAccessProfile{
+				DisableRunCommand: utils.Bool(!d.Get("run_command_enabled").(bool)),
+			}
+		}
 	}
 
 	if d.HasChange("auto_scaler_profile") {

